### PR TITLE
Add dockerfile for Reason

### PR DESCRIPTION
**Pull request by kayceesrk**
Also, you are most likely doing seamless cross-compilation without knowing it :-)

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM ocaml/opam:debian
+RUN sudo -u opam sh -c "opam depext -u merlin utop"
+RUN opam remote add main https://opam.ocaml.org && \
+    opam pin add -y merlin https://github.com/the-lambda-church/merlin.git#reason-0.0.1 && \
+    opam pin add -y merlin_extend https://github.com/let-def/merlin-extend.git#reason-0.0.1 && \
+    opam pin add -y reason https://github.com/facebook/reason.git#0.0.6
```
Comment (yunxing) [inline docker/Dockerfile]: I really need to write a script to auto-populate the tag.

**Comment by yunxing** [comment]
Looks awesome! Can't wait to try it out when I get back tonight.

**Comment by cristianoc** [comment]
Thanks! Works smoothly for me.

**Comment by kayceesrk** [comment]
@cristianoc Excellent!

**Pull request merged**